core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -88,31 +88,31 @@ private DataTask task(StaticTableScan scan) {
     return StaticDataTask.Row.of(partition.key, partition.recordCount, partition.fileCount);
   }
 
-  private Iterable<Partition> partitions(StaticTableScan scan) {
+  private static Iterable<Partition> partitions(StaticTableScan scan) {
     CloseableIterable<FileScanTask> tasks = planFiles(scan);
 
-    PartitionMap partitions = new PartitionMap(table().spec().partitionType());
+    PartitionMap partitions = new PartitionMap(scan.table().spec().partitionType());
     for (FileScanTask task : tasks) {
       partitions.get(task.file().partition()).update(task.file());
     }
     return partitions.all();
   }
 
   @VisibleForTesting
-  CloseableIterable<FileScanTask> planFiles(StaticTableScan scan) {
+  static CloseableIterable<FileScanTask> planFiles(StaticTableScan scan) {
+    Table table = scan.table();
+    Snapshot snapshot = table.snapshot(scan.snapshot().snapshotId());
```

Comment:
Slight refactor here to avoid looking up the same snapshot twice. This replaces the variable for `snapshotId`.